lib/_emerge/depgraph.py

```diff
@@ -9157,6 +9162,48 @@ def display_problems(self):
 			show_mask_docs()
 			writemsg("\n", noiselevel=-1)
 
+		# Build a list of problematic packages
+		maintainer_nag_packages = []
+
+		for pkg in self._dynamic_config.digraph:
+			if not isinstance(pkg, Package):
+				continue
+
+			if not pkg.type_name == 'ebuild':
+				continue
+
+			maintainer_nag = pkg.root_config.settings.repositories[pkg.repo].maintainer_nag
+			maintainer_nag_url = pkg.root_config.settings.repositories[pkg.repo].maintainer_nag_url
+
+			if not maintainer_nag:
+				continue
+
+			path = os.path.dirname(pkg.root_config.trees['porttree'].dbapi.findname(pkg.cpv, myrepo=pkg.repo))
+			path = os.path.join(path, 'metadata.xml')
+
+			if MetaDataXML is not None and os.path.isfile(path):
+				try:
+					metadata_xml = MetaDataXML(path, None)
+					maintainer = metadata_xml.format_maintainer_string()
+				except SyntaxError:
+					continue
+
+				if not maintainer:
+					maintainer_nag_packages.append(pkg.cpv + "::" + pkg.repo)
+
+		if maintainer_nag_packages:
+			writemsg("\n" + colorize("WARN", "!!!") + \
+				 " The following packages have no maintainer. They are at greater risk of bugs." +
+				 "\n!!! You can help to keep it in tree and fix bugs (inc. possible security issues)")
+
+			if maintainer_nag_url:
+				writemsg("\n!!! Please see: " + maintainer_nag_url)
+
+			print()
+			for pkg in maintainer_nag_packages:
+				print("- " + pkg)
```

Comment:
Since writemsg outputs to stderr, we need the rest of the output to go to stderr as well, otherwise there will be odd cases where the outputs do not appear together as they should.